### DISCUSSION TITLE
API별 인증 토큰 유형 관련 질문입니다 ㅠ

### DISCUSSION THREAD

**Discussion by e34652** [commerce-api-naver/commerce-solutions#100] [묻고 답하기]
https://github.com/commerce-api-naver/commerce-api/discussions/780#discussion-5304047
제가 위의 Q&A를 통해 이해한 내용이 맞는지 확인을 받고 싶습니다 ㅠ 

내 스토어 애플리케이션은 스토어 판매자 자격으로도 모든 API 호출이 가능하기에 
토큰은 SELF만 있으며 이는 판매자 권한을 의미합니다

또한 1 : 1로 연결되어 account_uid를 파라미터로 입력할 필요가 없습니다

ㅡㅡㅡㅡㅡ

솔루션 애플리케이션은 솔루션 자격이 필요한 API가 구분되기에
SELF(솔루션)와 SELLER(스토어 판매자)를 구분하여 토큰을 발급하고 

또한 1 : N으로 연결되기에  SELLER 토큰 발급 시
account_uid를 파라미터로 지정하고, roleGroupType와 대응하는 권한을 획득합니다

반면 SELF토큰 발급 시에는 account_uid 값이 필요 없으며 ("") 솔루션 자격 권한을 획득합니다

제가 이해한 내용이 맞을까요?

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

이해하신 내용이 맞습니다.
인증 토큰 발급 과정에서의 `type=SELF`는 발급 과정에서 사용한 애플리케이션의 종류에 따른 상대적인 개념입니다.
- 내 스토어 애플리케이션 사용: `type=SELF`는 "내 스토어"를 지칭함
- 솔루션 애플리케이션 사용: `type=SELF`는 대상하는 "솔루션"을 지칭함

커머스API 플랫폼에서 '인증 토큰'은 해당 사용자에 대한 API 호출 적격성 여부 등을 판단하게 됩니다.
이때 해당 '인증 토큰'에 따라 대상 API가 접근할 수 있는 정보의 범위가 달라집니다.
- `type=SELF`(솔루션): 특정한 API가 해당 솔루션에 대한 정보만 접근 가능
- `type=SELLER&account_id={A스토어의 UID}` (구독자 스토어): 특정한 API가 A스토어에 대한 정보만 접근 가능
- `type=SELLER&account_id={B스토어의 UID}` (구독자 스토어): 특정한 API가 B스토어에 대한 정보만 접근 가능

따라서 각각의 인증 토큰은 모두 개별적으로 관리될 수 있어야 합니다.

**Reply by e34652** [reply]
명쾌한 답변 감사합니다 선생님 
새해 복 많이 받으세요~

**Reply by commerce-api-naver** [reply]
@e34652 님, 커머스API 이민석입니다.

감사합니다.
담당자 분께서도 새해 복 많이 받으시기 바랍니다.
